eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -9,11 +9,28 @@ param (
 )
 Set-StrictMode -Version 3
 
-Install-Module -Name powershell-yaml -RequiredVersion 0.4.7 -Force -Scope CurrentUser
-
 . $PSScriptRoot/ChangedFiles-Functions.ps1
 . $PSScriptRoot/Logging-Functions.ps1
 
+$script:psYamlInstalled = $false
+function Ensure-PowerShell-Yaml-Installed {
```

Comment:
I'm not opposed to this but we might want to make this more generic like we have in tools https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Helpers/PSModule-Helpers.ps1